src/main/java/net/sf/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -48,14 +50,36 @@
      */
     Parser getParser();
 
+    /**
+     * Performs a cleanup of the fetched entry.
+     *
+     * Only systematic errors of the fetcher should be corrected here
+     * (i.e. if information is consistently contained in the wrong field or the wrong format)
+     * but not cosmetic issues which may depend on the user's taste (for example, LateX code vs HTML in the abstract).
+     *
+     * Try to reuse existing {@link Formatter} for the cleanup. For example,
+     * {@code new FieldFormatterCleanup(FieldName.TITLE, new RemoveBracesFormatter()).cleanup(entry);}
+     *
+     * By default, no cleanup is done.
+     * @param entry the entry to be cleaned-up
+     */
+    default void doPostCleanup(BibEntry entry) {
+        // Do nothing
+    }
+
     @Override
     default List<BibEntry> performSearch(String query) throws FetcherException {
         if (StringUtil.isBlank(query)) {
             return Collections.emptyList();
         }
 
         try (InputStream stream = new BufferedInputStream(getQueryURL(query).openStream())) {
-            return getParser().parseEntries(stream);
+            List<BibEntry> fetchedEntries = getParser().parseEntries(stream);
+
+            // Post-cleanup
+            fetchedEntries.forEach(this::doPostCleanup);
```

Comment:
Why is the implementor of `SearchBasedParserFetcher` required to do the cleanup. Why isn't there a fetcher framework which calls `performSearch` and after successfull fetching `doPostCleanup`?